### PR TITLE
front: fix various bugs related to PSLs edition

### PR DESCRIPTION
**Pull request by jacomyal**
This PR contains 3 commits that fix tickets:
  - [x] #4975 
  - [x] #4982 
  - [x] #5789 
  - [x] #5790

### CODE REVIEW

**Review by flomonster — COMMENTED** [review]
I've tested it and found out few bugs/irritants

- **Bug**: We cannot save a newly created PSL with panels. Editoast expects some kp but we cannot provide them through the form.
- **Bug**: When editing a speed limit some track sections can't be added or removed from the speed limit. I didn't find the reasons why these tracks couldn't be selected.
- **Irritant**: There is no way to see which speed section is currently edited.

Note: I don't know if these points need to be fixed by this PR. We can open new tickets if you wish.

File: front/src/common/Map/Layers/GeoJSONs.tsx [hunk 1]
```diff
@@ -85,12 +120,23 @@ function adaptFilter(
...
-  if (whiteList.length) conditions.push(['in', fieldId, ...whiteList]);
-  if (blackList.length) conditions.push(['!in', fieldId, ...blackList]);
+  if (hasExpressionFilters) {
+    // Add the conditions as ExpressionFilterSpecification:
+    if (whiteList.length) conditions.push(['in', ['get', fieldId], ['literal', whiteList]]);
+    if (blackList.length) conditions.push(['!', ['in', ['get', fieldId], ['literal', blackList]]]);
+  } else {
+    // Add the conditions as LegacyFilterSpecification:
+    if (whiteList.length) conditions.push(['in', fieldId, ...whiteList]);
+    if (blackList.length) conditions.push(['!in', fieldId, ...blackList]);
```
Comment (SharglutDev) [inline front/src/common/Map/Layers/GeoJSONs.tsx]: It's not related to your PR but as we are on a open source project (and anyhow), I think we should use another syntax than "whitelist" and "blacklist"

**Comment by jacomyal** [comment]
> **Bug**: We cannot save a newly created PSL with panels. Editoast expects some kp but we cannot provide them through the form.

New ticket #5792 will be handled outside this PR, since it involves back as well.

> **Bug**: When editing a speed limit some track sections can't be added or removed from the speed limit. I didn't find the reasons why these tracks couldn't be selected.

Ticket #5800 will be added to this PR if I find a fix before it's reviewed.

> **Irritant**: There is no way to see which speed section is currently edited.

Ticket #5799 will be added to this PR if I find a fix before it's reviewed.

**Review by SharglutDev — CHANGES_REQUESTED** [review]
Thank you for the PR.

Left one comment and you have some warnings left in the code.

**Review by SharglutDev — APPROVED** [review]
Thank you for the changes and comments on the function. Lgtm !

**Review by clarani — APPROVED** [review]
LGTM, tested :white_check_mark:

**Pull request merged**